### ISSUE TITLE
Error about missing "archive.info" locally when doing S3-only backup

### ISSUE THREAD

**Issue by sagargp** [pgbackrest/pgbackrest#2388]
1. pgBackRest version: **2.5.2**

2. PostgreSQL version: **16.3**

3. Operating system/version: **Debian Bookworm (Docker image)**

4. Did you install pgBackRest from source or from a package? **Package**

5. Please attach the following as applicable:

**pgbackrest.conf**

```
[global]
archive-async=y
spool-path=/var/spool/pgbackrest
log-level-console=info
log-level-file=debug
start-fast=y

# Local storage
repo1-cipher-pass=
repo1-cipher-type=aes-256-cbc
repo1-path=/var/lib/pgbackrest
repo1-retention-full=2

# S3
repo2-type=s3
repo2-s3-bucket=demo-wal-archive-testing
repo2-s3-endpoint=s3.us-west-2.amazonaws.com
repo2-s3-key=
repo2-s3-key-secret=
repo2-s3-region=us-west-2
repo2-retention-full=4
# override the following with --repo2-path=/wal-archives/$hostname
# repo2-path=/wal-archives/

[demo]
pg1-path=/var/lib/postgresql/data
pg1-database=plangraph
pg1-user=postgres

[global:archive-push]
compress-level=9
process-max=4

[global:archive-get]
process-max=4
```

With the previous config, I have done a `stanza-create` successfully, and see the stanza data on my local storage (`/var/lib/pgbackrest` and also in S3. However, immediately after I tried to perform a `backup` only on repo2:

```
pgbackrest --stanza=demo --repo=2 --repo2-path='/wal-archives/myhostname' backup
```

This times out after a while, and I get the following in the Postgres logs:

```
ERROR: [104]: raised from local-1 protocol: archive-push command encountered error(s):
       repo2: [FileMissingError] unable to load info file '/var/lib/pgbackrest/archive/demo/archive.info' or '/var/lib/pgbackrest/archive/demo/archive.info.copy':
       FileMissingError: unable to open missing file '/var/lib/pgbackrest/archive/demo/archive.info' for read
       FileMissingError: unable to open missing file '/var/lib/pgbackrest/archive/demo/archive.info.copy' for read
       HINT: archive.info cannot be opened but is required to push/get WAL segments.
       HINT: is archive_command configured correctly in postgresql.conf?
       HINT: has a stanza-create been performed?
       HINT: use --no-archive-check to disable archive checks during backup if you have an alternate archiving scheme.
           [CommandError] on retry at 32ms: archive-push command encountered error(s):
       repo2: [FileMissingError] unable to load info file '/var/lib/pgbackrest/archive/demo/archive.info' or '/var/lib/pgbackrest/archive/demo/archive.info.copy':
       FileMissingError: unable to open missing file '/var/lib/pgbackrest/archive/demo/archive.info' for read
       FileMissingError: unable to open missing file '/var/lib/pgbackrest/archive/demo/archive.info.copy' for read
       HINT: archive.info cannot be opened but is required to push/get WAL segments.
       HINT: is archive_command configured correctly in postgresql.conf?
       HINT: has a stanza-create been performed?
       HINT: use --no-archive-check to disable archive checks during backup if you have an alternate archiving scheme.
```

Does this mean a backup on local vs. S3 can't be done independently? I must do them both in synchrony?

Edit: Actually, more mysteriously, the path it's complaining about does exist: `/var/lib/pgbackrest/archive/demo/archive.info` -- which got created by the `stanza-create`.

**Comment by pgstef** [comment]
Hi,

`repo2: [FileMissingError] unable to load info file '/var/lib/pgbackrest/archive/demo/archive.info'` it actually complains about repo2-path. You haven't set it in your configuration, so by default it is `/var/lib/pgbackrest` in your S3 bucket...

`--repo2-path='/wal-archives/myhostname'` you shouldn't put that in your backup command. The backup heavily relies on WAL archives which are pushed by PG using the archive-push command, and that command needs the information from the configuration file.

In summary, put `repo2-path='/wal-archives/myhostname'` in your configuration file, run the `stanza-create` again, then run the `check` command to verify that WAL archiving is working to both repositories and then you can take backups individually for each repository if you don't mess with the repo settings.

Kind Regards

**Comment by sagargp** [comment]
I want the `repo2-path` to be dynamic (eg., `/wal-archives/$HOSTNAME` for whatever host it's running on). I assumed that `--repo2-path` would override this per-command. Is that not right? I did pass it as an argument when I did `stanza-create` and I see the stanza data on my S3 host in the expected path.

**Comment by dwsteele** [comment]
I don't think this makes much sense:
```
I want the repo2-path to be dynamic (eg., /wal-archives/$HOSTNAME
```
The repo/stanza should be per Postgres cluster not per host.

> I tried what you said (omitting --repo2-path from the CLI

I think you missed something, then, but without logs I couldn't say what. `/var/lib/pgbackrest` is the default that pgBackRest uses when `repoX-path` is not set.

I don't see any pgBackRest issues here so I'm going to close this. I recommend you have a look at the user guide (https://pgbackrest.org/user-guide.html) and tailor your solution accordingly.

**Comment by sagargp** [comment]
Our use case is such that every host has is a full stack deployment of our application, and has a Postgres instance on it that is totally independent of all others. We are testing `pgbackrest` for point-in-time recoveries onto separate simulation hosts so that we can do postmortem debugging much more easily, along with regression testing. It is an unusual use case, and we are still in the exploration phase.

Meanwhile, I think I figured out what I was doing wrong, for anyone else who comes across this thread. It boils down to two points:

* It seems like `pgbackrest` inconsistently applies the `--repoN-path` argument, and for some commands (such as `stanza-create`) this argument is respected, while other commands (such as `backup`) this argument is ignored in favor of what's in the configuration file.
* I did not realize that changing the `pgbackrest.conf` file requires Postgres to restart.

Thus my trial-and-error approach to playing with these commands yielded inconsistent results. I started over from scratch and this time I omitted the `repo2-path` from the config file, and made sure to consistently use the `--repo2-path` argument on the command line, and everything seems to be working consistently now.

**Comment by pgstef** [comment]
> Our use case is such that every host has is a full stack deployment of our application, and has a Postgres instance on it that is totally independent of all others. We are testing `pgbackrest` for point-in-time recoveries onto separate simulation hosts so that we can do postmortem debugging much more easily, along with regression testing. It is an unusual use case, and we are still in the exploration phase.

There's a missing piece in your explanation (outside of configurations obviously). Where are the backups taken from ?
In the configuration that you provided in your initial config, there's only 1 Postgres host. So I don't quite follow why it would be a problem to have 1 config file on the host with proper repo path...

If you're running the backups from a remote dedicated backup server that points to multiple Postgres hosts, then you could simply have one stanza per Postgres host and not mess with command line options. That's what repo hosts are made for. 

>     * It seems like `pgbackrest` inconsistently applies the `--repoN-path` argument, and for some commands (such as `stanza-create`) this argument is respected, while other commands (such as `backup`) this argument is ignored in favor of what's in the configuration file.
> 
>     * I did not realize that changing the `pgbackrest.conf` file requires Postgres to restart.

That's totally wrong assumptions. You're hoping that `backup --repoX-path` will apply the same cmd line setting to the `archive-push` which is run by Postgres in background of the backup. So it has no way to figure that out. That's why you should not overwrite repo configuration in the command line (to be honest I think we should remove that possibility for some settings).

As you can read [here](https://pgbackrest.org/command.html#introduction): _Command-line options override environment options which override config file options. _

When changing the `archive_command`, you have to reload Postgres configuration, not restart. And if you change the config file, you don't have to restart Postgres at all. But the config change will only be applied next time Postgres calls the archive-push command ; all the running processes will not be updated indeed. By restarting Postgres, you've forced to kill those running processes and start new ones - which you could have done with the pgBackRest stop command.

> Thus my trial-and-error approach to playing with these commands yielded inconsistent results. I started over from scratch and this time I omitted the `repo2-path` from the config file, and made sure to consistently use the `--repo2-path` argument on the command line, and everything seems to be working consistently now.

Yes, if you provide the repo2-path on the command line for every command, this will work. But again, you shouldn't do that. The repo should be the same for all the stanzas defined in the configuration file. What you should do it to use different stanza names, so you'll have a repository per stanza in your repo. Which would achieve exactly what you would have wanted at the first place without trying to add unnecessary complication.
